### ISSUE TITLE
diffuser le direct émissions

### ISSUE THREAD

**Issue by bioboosterbob** [3forges/yellowradio_proposals#1]
utiliser le player twitch

**Comment by bioboosterbob** [comment]
aussi: https://github.com/koala-interactive/wowza-webrtc-player

**Comment by bioboosterbob** [comment]
aussi:
https://gcore.com/pricing/streaming-platform
![image](https://github.com/3forges/yellowradio_proposals/assets/89749441/f89f007d-251b-4b80-95ac-63983e1d6943)

**Comment by bioboosterbob** [comment]
aussi:
https://flashphoner.com/player-for-browser/
![image](https://github.com/3forges/yellowradio_proposals/assets/89749441/a97f7880-99f9-4943-9bdf-c3cbd59e1e82)
![image](https://github.com/3forges/yellowradio_proposals/assets/89749441/d63f82a2-56ac-49d5-bd9c-3ca968d84271)

**Comment by bioboosterbob** [comment]
aussi:
https://reference.dashif.org/dash.js/latest/samples/index.html#Live

**Comment by Jean-Baptiste-Lasselle** [comment]
> solves #1
> 
> La tâche est :
> 
> * d'ajouter le player video et le chat Twitch dans la page d'accueil comme sur https://radiojaune.com
> * de le faire en faisant un composant react, masi dans le coontexte preact (npx astro add preact)
> * du react pour pouvoir hydrater le composant avec une hydratation `client:visible`
> 
> JB :
> 
> > Du coup pour utiliser une balise script dans un composant react, c'est trop compliqué pour l'instant,
> > On oublies l'idée de faire un composant react pour el twitch player, on utilises juste un *.astro
> 
> ![image](https://user-images.githubusercontent.com/35227860/258952862-f9e46756-cd56-4114-b665-50d98f42dbad.png)

@bioboosterbob j'ai git push une version que j'ai réussi à faire tu verras sur la branche `jb/twitch/player`

l'idée de faire un composant react n'est pas mauvaise, mais par contre là on a pas le temps, il nous faut un site simple efficace

et tu verras on aura bien assez de travail avec ce qu'ils vont nous demander, on verra quand et si on fait des essais avec React à un autre moment

ça demande plus de connaissances en React, on peut pas faire un composant juste comme ça sinon ça va défoncer notre performance

ne pas confondre un labo et le champs de bataille pour de vrai

ps: et donc pour l'instant ok on a compris l'hydratation, mais la mettre en oeuvre sur le composant twitch ça sera pas tt de suite.

### Mais aussi

J'ai créé https://github.com/3forges/yellowradio_proposals/issues/7 pour une tâche plus tard un jour sur les composants react ou preact dans le projet Astro

**Comment by Jean-Baptiste-Lasselle** [comment]
> aussi: https://reference.dashif.org/dash.js/latest/samples/index.html#Live ![image](https://user-images.githubusercontent.com/89749441/258579556-b155d7e7-2794-4afe-b783-2fab670d1a86.png)
> 
> aussi: dans le src , on as des 'lupus lorem ...' stream disponibles

nope, on utilisera pas dash: parce que si Twitch fait son propre client, c'est parce que l'API Twitch évolue rapidement, et il doivent faire leur propre client pour qu'il existe au moins un composant sur le marché open source, qui fonctionne pour avoir un player twitch, et c'est la même pour toutes les autres plateformes

**Comment by Jean-Baptiste-Lasselle** [comment]
> aussi: https://flashphoner.com/player-for-browser/ ![image](https://user-images.githubusercontent.com/89749441/258579116-a97f7880-99f9-4943-9bdf-c3cbd59e1e82.png) ![image](https://user-images.githubusercontent.com/89749441/258579162-d63f82a2-56ac-49d5-bd9c-3ca968d84271.png)
> 
> [edited] ![image](https://user-images.githubusercontent.com/89749441/258583629-dddfe1b1-9c88-4f3c-9b3c-c15fe625afbc.png) on oubli

Cette offre dont tu parles et toutes les précédentes, cela n'a rien à voir avec ce qu'on cherche à faire, c'est autre chose ce dont ils parlent, les services de streaming en question c'est côté serveur, pas côté client comme nous: 
ça servirai à rien d'utiliser un(des) serveur(s) supplémentaire, entre notre client, et les serveurs twitch, peu importe où il(s) serai(en)t plac(é)s par rapport au browser et aux serveurs twitch
